### DISCUSSION TITLE
Abort stale commit/pool requests

### DISCUSSION THREAD

**Discussion by hamannjames** [livewire/livewire#9030] [3. Feature ideas]
When multiple changes are queued up to be sent to the server, we often only want to process the latest one. Take a typeahead input as an example. If I am searching U.S. states in the backend, and a user types in "Al," I would perform a search and show them Alabama and Alaska. But if the user deletes the "l" and types in "r" I want to show them Arkansas. The problem is that while they will eventually only see Arkansas, they might see Alabama and Alaska flash on the screen before then. Also, if for some reason the "Ar" search takes a while, they might be staring at stale data for a long time.

Using the [Abort Controller API](https://developer.mozilla.org/en-US/docs/Web/API/AbortController), I was able to get a fun solution working for this by extending core. If the user wanted, they could use a magic "$abort" Alpine method to cancel the previous request containing that specific component in the payload. This was done by hooking into the "request" event and using a Bag() to keep track of components and abort controllers. Not only did this prevent a flash of stale content, it also sped up response time, because Livewire will push the next change to be processed on a fail, which a abort triggers natively. 

Unfortunately, my solution didn't work if multiple stale changes are waiting to be sent. For example, if there is a button in the component that calls a slow backend request, and the user clicks it multiple times, the next time the user calls $abort it will only abort the first change queued, because technically the request for other changes have not been pooled (committed, whatever the right term is). I am still searching for a good place to hook into the commit process to fix this. I thought if I could access the request bus on abort and remove commits for a specific component that would work, but it doesn't. There is some place where changes live in limbo between a user interaction and hitting the bus that I am missing.

I wanted to make this post to put this idea forward and maybe get some guidance. Like I said, I was able to get half a solution working just by adding a new feature to core and using existing hooks. I would imagine it wouldn't take much more work to get a good user API working for this.

**Comment by CapitaineToinon** [comment]
Hey, would you mind share the code you wrote to ignore stale requests? I was trying to do the exact same thing but I'm just not there yet. Personally I think this should be handled by livewire automatically but having a workaround would be neat.
